src/%IPM/Main.cls

```diff
@@ -991,9 +1155,44 @@ ClassMethod Init(ByRef pCommandInfo) [ Internal ]
 	Set tSCClassChanged = 0
 	
 	Set tPrimaryClass = ""
-	Set tCompositeSCClass = "%IPM.PackageManager.Developer.Extension.Composite"
-	Set tPackageMgrSCClass = "%IPM.PackageManager.Developer.Extension.PackageManager"
-	Set tSCInterface = "%IPM.PackageManager.Developer.Extension.SourceControl.Interface"
+	Set tCompositeSCClass = "%IPM.ExtensionBase.Composite"
+	Set tServerMenuExtClass = ""
+
+	// set server menu extension class (tServerMenuExtClass)
+	If tQuiet {
+		If $$$HasModifier(pCommandInfo,"menuextension") {
```

Comment:
Hi Dmitry! This is a case where the user initialize IPM with a full command including server menu extension, which will probably be a very common use case for our studio/vscode developers. Besides, the menu extension class is a very generic class for those who want to set server menu extension, and we've provided such as dummy class as the super class: %IPM.ExtensionBase.ServerMenu.Interface. Therefore, I think is should be part of IPM.

Reply:
Good point and I agree! However, this is generic server menu extension, not Studio extension, which I believe is generic enough to keep it in IPM.
@isc-kiyer  can weigh in

Reply:
IPM includes an IDE (not Studio-specific) extension to help with Package Manager-specific stuff. This (and related classes about which you had questions) support enabling multiple extensions at the same time without needing to write a separate extension to manage all of them. Ideally this would be in IRIS but since it isn't we can't exactly put it in a dependency of the package manager. (I think we kept the package manager extension, right?)

Reply:
Clarifying on what I believe we've agreed on: Specific source control extension implementations (such as ISC's related to Perforce) should be separate packages, not part of IPM itself. The package manager extension should be part of the package manager, as well as the ability to use it seamlessly with arbitrary source control extensions (and to make source control extensions aware of events from a package manager perspective that are relevant to them).

Reply:
@isc-tleavitt  Thanks for clarifying! Yes, specific-source-control should be in independent package, for example zpm-perforce. But we do provide interface for those (%IPM.ExtensionBase.SourceControl.Interface) which should be fine.